README.md

```diff
@@ -62,7 +62,11 @@ in the following manner:
 CastleApi newAPIRef = Castle.sdk().onRequest(req)
 ```
 
-Here `req` is an instance of `HttpServletRequest`.
+Here `req` is an instance of `HttpServletRequest`. 
+
+Note that the `req` instance should be bind to the underlaing request in order to extract the necessary information. 
+It means that a safe place to create the `CastleApi` instance is the request handling thread. After creation the 
+`CastleApi` instance can be passed to any thread independently of the original thread life cycle.
```

Comment:
Additional information about async use of the CastleApi. I do not know if we should be more explicit here.